napari/_qt/dialogs/_tests/test_preferences_dialog.py

```diff
@@ -280,3 +283,70 @@ def test_preferences_dialog_restore(qtbot, pref, monkeypatch):
             ).text()
         )
     ) == KeyBinding.from_str('Ctrl')
+
+
+@skip_local_focus
+@skip_on_mac_ci
+@pytest.mark.key_bindings
+@pytest.mark.parametrize(
+    'confirm_key',
+    ['enter', 'return', 'tab', 'capslock'],
+)
+def test_preferences_dialog_not_dismissed_by_keybind_confirm(
+    qtbot, pref, confirm_key
+):
+    """This test ensures that when confirming a keybinding change, the dialog is not dismissed.
+
+    Notes:
+        * Skipped on macOS CI due to accessibility permissions not being
+          settable on macOS GitHub Actions runners.
+        * For this test to pass locally, you need to give the Terminal/iTerm/VSCode
+          application accessibility permissions:
+              `System Settings > Privacy & Security > Accessibility`
+
+        See https://github.com/asweigart/pyautogui/issues/247 and
+        https://github.com/asweigart/pyautogui/issues/247#issuecomment-437668855
+    """
```

Comment:
This docstring is badass @psobolewskiPhD 😍

Reply:
Its 100% stolen from @dalthviz 
Similar test in `napari/_qt/widgets/_tests/test_shortcut_editor_widget.py`